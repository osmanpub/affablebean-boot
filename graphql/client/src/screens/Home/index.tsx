import React from "react";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Categories as CategoriesState } from "../../interfaces/categories";
import { Match } from "../../interfaces/router";

type Props = {
  categories: CategoriesState;
  match: Match;
};

function Home(props: Props) {
  const { match } = props;
  const url = match ? match.url : "";

  return (
    <div>
      <Header url={url} />
      {<Categories />}
      <Footer />
    </div>
  );
}

export default Home;
