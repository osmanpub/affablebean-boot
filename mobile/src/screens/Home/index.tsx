import React, {useEffect} from 'react';
import {ScrollView} from 'react-native';
import {connect, useDispatch} from 'react-redux';
import Categories from '../../components/Categories';
import Header from '../../components/Header';
import {Cart} from '../../interfaces/cart';
import {fetchCategoriesIfNeeded} from '../../net/categories';
import {RootState} from '../../redux';
import {Categories as CategoriesState} from '../../interfaces/categories';

type Props = {
  cart: Cart;
  categories: CategoriesState;
  setCategoryProduct: Function;
  setScreen: Function;
};

function Home(props: Props) {
  const {cart, categories, setCategoryProduct, setScreen} = props;
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCategoriesIfNeeded());
  }, []);

  const {items} = categories;

  if (items.length === 0) {
    return null;
  }

  return (
    <ScrollView>
      <Header cart={cart} currentScreen="Home" setScreen={setScreen} />
      <Categories categories={items} setCategoryProduct={setCategoryProduct} />
    </ScrollView>
  );
}

const mapStateToProps = (state: RootState) => ({
  cartegories: state.cart,
  categories: state.categories,
});

export default connect(mapStateToProps)(Home);
