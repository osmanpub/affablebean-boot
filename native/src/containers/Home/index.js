import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { ScrollView } from "react-native";
import Categories from "../../components/Categories";
import Header from "../../components/Header";
import { fetchCategoriesIfNeeded } from "../../net/categories";

class Home extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchCategoriesIfNeeded());
  }

  render() {
    const { cart, categories, match } = this.props;
    const { items } = categories;
    const url = match ? match.url : "";

    if (items.length === 0) {
      return null;
    }

    return (
      <ScrollView>
        <Header cart={cart} url={url} />
        <Categories categories={items} />
      </ScrollView>
    );
  }
}

const mapStateToProps = state => {
  const { cart, categories } = state;

  return {
    cart,
    categories
  };
};

export const ConnectedHome = connect(mapStateToProps)(Home);

Home.propTypes = {
  cart: PropTypes.object,
  categories: PropTypes.object.isRequired,
  match: PropTypes.object
};
