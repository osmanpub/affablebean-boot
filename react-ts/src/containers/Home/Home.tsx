import React, { Component } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { fetchCategoriesIfNeeded } from "../../net/categories";
import { Props } from "../../interfaces/props";

export class Home extends Component<Props> {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchCategoriesIfNeeded());
  }

  render() {
    const { cart, categories, match } = this.props;
    const { items } = categories;

    if (items.length === 0) {
      return null;
    }

    return (
      <div>
        <Header cart={cart} url={match.url} />
        <Categories categories={items} />
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state: Props) => {
  const { cart, categories } = state;

  return {
    cart,
    categories
  };
};

export default connect(mapStateToProps)(Home);
