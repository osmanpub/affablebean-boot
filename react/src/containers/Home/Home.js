import React, { Component } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { fetchCategoriesIfNeeded } from "../../net/categories";

export class Home extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchCategoriesIfNeeded());
  }

  render() {
    const { cart, categories, match } = this.props;
    const { items } = categories;

    if (items.length === 0) {
      return null;
    }

    return (
      <div>
        <span id="hello">hello</span>
        <Header cart={cart} url={match.url} />
        <Categories categories={items} />
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart, categories } = state;

  return {
    cart,
    categories
  };
};

export default connect(mapStateToProps)(Home);
