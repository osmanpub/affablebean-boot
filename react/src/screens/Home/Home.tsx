import React, { useEffect } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Props } from "../../interfaces/props";
import { fetchCategoriesIfNeeded } from "../../net/categories";
import { RootState } from "../../redux";

export function Home(props: Props) {
  useEffect(() => {
    const { dispatch } = props;
    dispatch(fetchCategoriesIfNeeded());
  }, []);

  const { cart, categories, match } = props;
  const { items } = categories;
  const url = match ? match.url : "";

  if (items.length === 0) {
    return null;
  }

  return (
    <div>
      <Header cart={cart} url={url} />
      <Categories categories={items} />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
  categories: state.categories
});

export default connect(mapStateToProps)(Home);
