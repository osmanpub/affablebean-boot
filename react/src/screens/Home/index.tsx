import React, { useEffect } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Categories as CategoriesState } from "../../interfaces/categories";
import { fetchCategoriesIfNeeded } from "../../net/categories";
import { RootState } from "../../redux";

type Props = {
  categories: CategoriesState;
  dispatch: Function;
  match: any;
};

function Home(props: Props) {
  const { categories, dispatch, match } = props;
  const { items } = categories;
  const url = match ? match.url : "";

  useEffect(() => {
    dispatch(fetchCategoriesIfNeeded());
  }, [dispatch]);

  if (items.length === 0) {
    return null;
  }

  return (
    <div>
      <Header url={url} />
      <Categories categories={items} />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
  categories: state.categories
});

export default connect(mapStateToProps)(Home);
