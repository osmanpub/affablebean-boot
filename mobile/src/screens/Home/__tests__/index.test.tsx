import {render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Home from '..';

const mockStore = configureStore([thunk]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        product: {
          id: 1,
          description: 'semi skimmed (1L)',
          name: 'milk',
          price: 1.7,
        },
        quantity: 1,
        total: 1.7,
      },
    ],
    numberOfItems: 1,
    subtotal: 1.7,
  },
  categories: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        id: 1,
        name: 'dairy',
      },
      {
        id: 2,
        name: 'meats',
      },
      {
        id: 3,
        name: 'bakery',
      },
      {
        id: 4,
        name: 'fruit & veg',
      },
      {
        id: 5,
        name: 'cereals',
      },
      {
        id: 6,
        name: 'drinks',
      },
    ],
  },
};

const store = mockStore(initialState);

const home = (
  <Provider store={store}>
    <Home setCategoryProduct={jest.fn()} setScreen={jest.fn()} />
  </Provider>
);

describe('<Home />', () => {
  it('renders correctly', () => {
    const component = renderer.create(home).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('show logo', () => {
    const {getByTestId} = render(home);
    const logo = getByTestId(/logo/i);
    expect(logo).toBeTruthy();
  });

  it('show view cart link', () => {
    const {getByTestId} = render(home);
    const viewCart = getByTestId(/view-cart/i);
    expect(viewCart).toBeTruthy();
  });

  it('show clear cart link', () => {
    const {getByTestId} = render(home);
    const clearCart = getByTestId(/clear-cart/i);
    expect(clearCart).toBeTruthy();
  });

  it('show checkout link', () => {
    const {getByTestId} = render(home);
    const checkout = getByTestId(/checkout/i);
    expect(checkout).toBeTruthy();
  });

  it('show 1 cart items', () => {
    const {getByText} = render(home);
    const items = getByText(/1 items/i);
    expect(items).toBeTruthy();
  });

  it('show six categories', () => {
    const {getAllByTestId} = render(home);
    const boxes = getAllByTestId(/categoryBox/i);
    expect(boxes).toHaveLength(6);
  });
});
