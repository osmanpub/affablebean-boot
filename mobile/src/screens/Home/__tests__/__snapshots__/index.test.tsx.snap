// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/components/Header/logo.jpg",
          }
        }
        style={
          Object {
            "margin": 16,
          }
        }
        testID="logo"
      />
    </TouchableWithoutFeedback>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingBottom": 16,
          }
        }
      >
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          underlayColor="black"
        >
          <Text
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 32,
                "borderWidth": 1,
                "color": "blue",
                "fontWeight": "bold",
                "marginHorizontal": 8,
                "padding": 12,
              }
            }
            testID="clear-cart"
          >
            clear cart
          </Text>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          underlayColor="black"
        >
          <Text
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 32,
                "borderWidth": 1,
                "color": "blue",
                "fontWeight": "bold",
                "marginHorizontal": 8,
                "padding": 12,
              }
            }
            testID="view-cart"
          >
            view cart
          </Text>
        </TouchableHighlight>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 32,
                "borderWidth": 1,
                "color": "blue",
                "fontWeight": "bold",
                "marginHorizontal": 8,
                "padding": 12,
              }
            }
            testID="checkout"
          >
            checkout
          </Text>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 16,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/components/Header/cart.gif",
            }
          }
        />
        <Text
          testID="cart-items"
        >
           
          1
           items
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <ScrollView
      ListFooterComponent={
        <View
          style={
            Object {
              "padding": 64,
            }
          }
        />
      }
      data={
        Array [
          Object {
            "id": 1,
            "name": "dairy",
          },
          Object {
            "id": 2,
            "name": "meats",
          },
          Object {
            "id": 3,
            "name": "bakery",
          },
          Object {
            "id": 4,
            "name": "fruit & veg",
          },
          Object {
            "id": 5,
            "name": "cereals",
          },
          Object {
            "id": 6,
            "name": "drinks",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            dairy
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/dairy.jpg",
                }
              }
              testID="img-dairy"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            meats
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/meats.jpg",
                }
              }
              testID="img-meats"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            bakery
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/bakery.jpg",
                }
              }
              testID="img-bakery"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            fruit & veg
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/fruitveg.jpg",
                }
              }
              testID="img-fruit & veg"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            cereals
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/cereals.jpg",
                }
              }
              testID="img-cereals"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
            }
          }
          testID="categoryBox"
        >
          <Text>
            drinks
          </Text>
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/categories/drinks.jpg",
                }
              }
              testID="img-drinks"
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "padding": 64,
            }
          }
        />
      </View>
    </ScrollView>
  </View>
</SafeAreaView>
`;
